### PR TITLE
Download button for observation plans

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a download button for the observation plans.

### CODE REVIEW

File: skyportal/handlers/api/observation_plan.py [hunk 1]
```diff
@@ -200,3 +201,42 @@ def delete(self, observation_plan_request_id):
...
+        observation_plan_request = ObservationPlanRequest.get_if_accessible_by(
+            observation_plan_request_id,
+            self.current_user,
+            mode="read",
+            raise_if_none=True,
+            options=options,
+        )
+        self.verify_and_commit()
+
+        return self.success(data=observation_plan_request)
```
Comment (guynir42) [inline skyportal/handlers/api/observation_plan.py:242]: What's the difference between this and the regular `ObservationPlanRequest` handler?

File: skyportal/tests/frontend/test_gcnevents.py [hunk 2]
```diff
@@ -215,10 +215,10 @@ def test_gcnevents_observations(
     driver.click_xpath(submit_button_xpath)
 
     # check that the executed observation table appears
-    driver.wait_for_xpath('//*[text()="84434604"]')
-    driver.wait_for_xpath('//*[text()="ztfr"]')
-    driver.wait_for_xpath('//*[text()="1.57415"]')
-    driver.wait_for_xpath('//*[text()="20.40705"]')
+    # driver.wait_for_xpath('//*[text()="84434604"]')
```
Comment (guynir42) [inline skyportal/tests/frontend/test_gcnevents.py:218]: commented lines

**Review by guynir42 — COMMENTED** [review]
I feel like maybe we can do the same using the existing `ObservationPlanRequestHandler` if given an optional path argument.

File: skyportal/handlers/api/observation_plan.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation_plan.py]: @guynir42 Great point!

File: skyportal/tests/frontend/test_gcnevents.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/tests/frontend/test_gcnevents.py]: Oops!

File: skyportal/tests/frontend/test_gcnevents.py [hunk 3]
```diff
@@ -319,7 +319,7 @@ def test_observationplan_request(driver, user, super_admin_token, public_group):
     driver.click_xpath(submit_button_xpath)
 
     # wait for the observation plan to complete
-    time.sleep(15)
+    time.sleep(30)
```
Comment (stefanv) [inline skyportal/tests/frontend/test_gcnevents.py:322]: Sleeps are a big no-no in the test suite!

**Review by stefanv — COMMENTED** [review]
Looks good other than for that sleep.

File: skyportal/tests/frontend/test_gcnevents.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/tests/frontend/test_gcnevents.py:322]: @stefanv what do you do with things you run in the IOLoops?

Comment (mcoughlin) [reply skyportal/tests/frontend/test_gcnevents.py:322]: @stefanv ah timeouts. one sec.

**Comment by stefanv** [comment]
Stéfan comment :man_dancing:

**Review by guynir42 — APPROVED** [review]
LGTM

File: skyportal/tests/frontend/test_shifts.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def test_super_user_post_shift(
     driver.click_xpath(submit_button_xpath)
 
     # check for dropdown shift
-    driver.wait_for_xpath(f'//span[text()[contains(.,"{form_name}")]]')
+    driver.wait_for_xpath(f'//span[text()[contains(.,"{form_name}")]]', timeout=30)
```
Comment (guynir42) [inline skyportal/tests/frontend/test_shifts.py:54]: This appears to still be flaky.

Comment (mcoughlin) [reply skyportal/tests/frontend/test_shifts.py:54]: yeah I did some fighting with this stuff on the ExecutedObservations PR, but not sure I have actually resolved this in reality.

Comment (guynir42) [reply skyportal/tests/frontend/test_shifts.py:54]: I'm getting a sense that this is always failing on the dropdown menus. 

Also, maybe mark it as flaky and make a note to come back to it once you figure out the underlying problem (this isn't directly related to this PR).

Comment (mcoughlin) [reply skyportal/tests/frontend/test_shifts.py:56]: Good idea. Just marked it.

**Comment by mcoughlin** [comment]
Superceded by https://github.com/skyportal/skyportal/pull/2728 due to great difficulty rebasing after recent commits.

**Pull request closed without merging**